pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -525,7 +549,16 @@ func (asw *actualStateOfWorld) AddPodToVolume(markVolumeOpts operationexecutor.M
 	}
 
 	podObj, podExists := volumeObj.mountedPods[podName]
-	if !podExists {
+
+	updateUncertainVolume := false
+	if podExists {
+		// Update uncertain volumes - the new markVolumeOpts may have updated information.
+		// Especially reconstructed volumes (marked as uncertain during reconstruction) need
+		// an update.
+		updateUncertainVolume = utilfeature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) && podObj.volumeMountStateForPod == operationexecutor.VolumeMountUncertain
+	}
+	if !podExists || updateUncertainVolume {
```

Comment:
I'm not sure I follow, can you please elaborate?

Reply:
So there was this bug - https://github.com/kubernetes/kubernetes/issues/103143 which was reported because during reconstruction we do not get correct `outerSpecName` and if populator loop did not add volume to DSOW because of api-server unavailability then we never properly mount the volume.

After your change - it looks like the entire data structure stored in ASOW is restored for uncertain volumes, so we still need `SyncReconstructedVolume` ?

Reply:
You're correct, with `SELinuxMountReadWriteOncePod` feature enabled it should not be necessary to call `SyncReconstructedVolume`. I added a new commit that removes it.